### ISSUE TITLE
Matching Sender

### ISSUE THREAD

**Issue by ArkaneMoose** [ArkaneMoose/BotBot#10]
BotBot should be able to use regexes to match the sender of a message.

**Comment by jedevc** [comment]
Do you have an idea of what this would look like in terms of syntax?

**Comment by ArkaneMoose** [comment]
I'm thinking something like the following:
`Sender-matching regex -> Message-matching regex -> Response`

**Comment by jedevc** [comment]
Hm. Maybe... It could get annoying if you ever needed to match anything else.

But at the moment it looks ok. But how would you keep code compatible? You'd have to allow for only one regex which would be automatically be the message matching one.

The syntax looks a little weird as well. 2 arrows? I think the arrow should be between the regexes and the response and another symbol should be used for between the regexes. Maybe "<->"?

**Comment by ArkaneMoose** [comment]
Yes, if there is only one regex (and therefore only one arrow), it would default to message-regex only.

`<->` could work, though I feel it is more intuitive to use another `->`.
